File: dart-funk/lib/src/parser_exception.dart
import 'package:monkeydart/monkeydart.dart';

class ParserException implements Exception {
  ParserException(
    this.message,
    this.parser,
    this.statements,
  );

  final String message;
  final Parser parser;
  final List<Statement> statements;

  @override
  String toString() {
    return '🙈-🙉-🙊: $message';
  }
}

File: java/monkey/src/main/java/root/parser/ParserException.java
package root.parser;

import root.LocalizedToken;
import root.util.ExceptionUtil;

public class ParserException extends Exception {

    private final LocalizedToken localizedToken;

    public ParserException(String message, LocalizedToken localizedToken) {
        super(ExceptionUtil.buildMessage("InvalidSyntax", message, localizedToken));
        this.localizedToken = localizedToken;
    }

    public LocalizedToken getLocalizedToken() {
        return localizedToken;
    }
}
